apps/web/src/templates/algo-address-template.tsx

```diff
@@ -0,0 +1,156 @@
+import { CollectibleWithDetails } from '@algomart/schemas'
+import useTranslation from 'next-translate/useTranslation'
+
+import css from './algo-address-template.module.css'
+
+import CollectibleBrowserDialog from '@/components/collectibles/collectible-browser-dialog'
+import CollectibleItem from '@/components/collectibles/collectible-item'
+import ExternalLink from '@/components/external-link'
+import Grid from '@/components/grid/grid'
+import Heading from '@/components/heading'
+import Pagination, { PAGE_SIZE } from '@/components/pagination/pagination'
+import Select, { SelectOption } from '@/components/select/select'
+import { useAuth } from '@/contexts/auth-context'
+import { formatAlgoAddress } from '@/utils/format-string'
+
+export interface AlgoAddressTemplateProps {
+  activeAsset: CollectibleWithDetails | null
+  algoAddress: string
+  algoAddressIsInvalid: boolean
+  assets: CollectibleWithDetails[]
+  currentPage: number
+  handlePageChange: (pageNumber: number) => void
+  handleSortChange: (option: SelectOption) => void
+  isViewerActive: boolean
+  selectedOption: SelectOption
+  selectOptions: SelectOption[]
+  toggleViewer: (asset?: CollectibleWithDetails | undefined) => void
+  totalCollectibles: number
+}
+
+export default function AlgoAddressTemplate({
+  activeAsset,
+  algoAddress,
+  algoAddressIsInvalid,
+  assets,
+  currentPage,
+  handlePageChange,
+  handleSortChange,
+  isViewerActive,
+  selectedOption,
+  selectOptions,
+  toggleViewer,
+  totalCollectibles,
+}: AlgoAddressTemplateProps) {
+  const { user } = useAuth()
+  const { t } = useTranslation()
+
+  const formattedAlgoAddress = formatAlgoAddress(algoAddress)
+  const isCurrentUser = user?.address === algoAddress
+
+  const Address = () => (
+    <>
+      <span className={css.isDesktop}>{algoAddress}</span>
+      <span className={css.isMobile}>{formattedAlgoAddress}</span>
+    </>
+  )
+
+  const Title = () => {
+    return (
+      <Heading className={css.pageTitle} level={1}>
+        {t('collection:algoAddressPage.Algorand Account')}:
+        <div>
+          {algoAddressIsInvalid ? (
+            <Address />
+          ) : (
+            <ExternalLink
+              href={`https://algoexplorer.io/address/${algoAddress}`}
```

Comment:
This is just an external link to AlgoExplorer, so I've hardcoded it to mainnet, which I think is fine as it won't impact any functionality _within_ our app. But just wanted to point it out.

Reply:
You're correct, it wouldn't work outside of production. This was half laziness, but also half not wanting to add another (duplicative) environment variable. The `configureAlgoExplorerUrl` lives within the API, not the web front end. I could absolutely do that, it just seems like a lot of overhead to have an external link resolve correctly in a non-production environment. I'll defer to the team's concensus.

Reply:
Could hardcode values based on the `ALGOD_ENV` value, but this is probably fine either way. There's a similar thing on the user's settings page.